neuraxle/metaopt/auto_ml.py

```diff
@@ -0,0 +1,358 @@
+        """
+        Find the next best hyperparams using previous trials.
+
+        :param trials_data_container: trials data container
+        :type trials_data_container: AutoMLDataContainer
+        :return: next best hyperparams
+        :rtype: HyperparameterSamples
+        """
+        raise NotImplementedError()
+
+    def fit_transform(self, data_inputs, expected_outputs=None) -> ('AutoMLStrategyMixin', float):
+        """
+        Fit cross validation with wrapped step, and return the score.
+
+        :param data_inputs: data inputs
+        :param expected_outputs: expected outputs to fit on
+        :return: self, step score
+        :rtype: (AutoMLStrategyMixin, float)
+        """
+        step = copy.copy(self.wrapped)
+        step: BaseCrossValidationWrapper = copy.copy(self.validation_technique).set_step(step)
+        step = step.fit(data_inputs, expected_outputs)
+
+        return self, step.get_score()
+
+    def is_higher_score_better(self):
+        return self.higher_score_is_better
+
+    def fit(self, data_inputs, expected_outputs=None):
+        return self
+
+    def transform(self, data_inputs, expected_outputs=None):
+        return data_inputs
+
+
+class RandomSearchBaseAutoMLStrategy(AutoMLStrategyMixin, MetaStepMixin, BaseStep):
+    """
+    Random Search Automatic Machine Learning Strategy that randomly samples the space of random variables.
+
+    .. seealso::
+        :class:`AutoMLStrategyMixin`,
+        :class:`AutoMLDataContainer`,
+        :class:`HyperparameterSamples`,
+        :class:`HyperparameterSpace`
+    """
+
+    def __init__(self, validation_technique=None):
+        AutoMLStrategyMixin.__init__(self, validation_technique)
+        MetaStepMixin.__init__(self, None)
+        BaseStep.__init__(self)
+
+    def find_next_best_hyperparams(self, trials_data_container: 'AutoMLDataContainer') -> HyperparameterSamples:
+        return trials_data_container.hyperparameter_space.rvs()
+
+
+class AutoMLDataContainer:
+    """
+    Simple data container used by :class:`AutoMLStrategyMixin`, and :class:`AutoMLSequentialWrapper`.
+    """
+
+    def __init__(
+            self,
+            trial_number: int,
+            scores: List[float],
+            hyperparams: List[HyperparameterSamples],
+            hyperparameter_space: HyperparameterSpace,
+            n_iters: int
+    ):
+        self.hyperparams = hyperparams
+        self.scores = scores
+        self.trial_number = trial_number
+        self.hyperparameter_space = hyperparameter_space
+        self.n_iters = n_iters
+
+
+class AutoMLSequentialWrapper(NonTransformableMixin, MetaStepMixin, BaseStep):
+    """
+    A step to execute any Automatic Machine Learning Algorithms.
+
+    Example usage :
+
+    .. code-block:: python
+
+        auto_ml = AutoMLSequentialWrapper(
+            auto_ml_strategy=RandomSearchAutoMLStrategy(),
+            step=ForecastingPipeline(),
```

Comment:
step could be first argument?

Reply:
Could you also give an example on how to use the Validation strategies into this step and unit test that combination of Validation wrapper + auto ml wrapper and all?

Reply:
added a unit test with a validation split wrapper as the validation technique